management_app/client_secret.json

```diff
@@ -0,0 +1,13 @@
+{
+  "web": {
+    "client_id": "559245110133-8bog3pa5k75ih9q1fkkauiv8latmbgjf.apps.googleusercontent.com",
+    "project_id": "teaching-management-demo",
+    "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+    "token_uri": "https://oauth2.googleapis.com/token",
+    "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+    "client_secret": "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",
+    "redirect_uris": [
+      "http://127.0.0.1:5000/auth/login_callback"
+    ]
+  }
+}
```

Comment:
Is it possible to have a `.env` file to store this sensitive information? Then we can share this Json config offline without publishing it in Github. There is a `.flaskenv` file to store the variable, you can also refer to that file to create a `.env` if needed.

Reply:
Tks for mentioning this. We are using credentials provided by the Google Cloud Platform to implement the uci login feature. From my understanding, the provider will look for a plaintext JSON file that contains particular fields like `redirect_uris` and `client_secret` and use it for verification on its end.  (https://developers.google.com/api-client-library/dotnet/guide/aaa_client_secrets)

The URLs we are using are local URLs and tokens are generated by my GCP account for local test use. So basically this data is just for the sake of the development phase. Once we successfully deployed the project on UCI servers, we will replace those tokens and URLs then as the final productionalizing step. And since the application is deployed on the UCI server, the JSON file that contains critical data should only be visible to the IT admins of UCI.

Reply:
Got it. Thanks for the explanation.